### PR TITLE
[8.0][product_multi_image] Multi images for products

### PR DESCRIPTION
**Pull request by yajo**
This will be WIP until https://github.com/OCA/server-tools/pull/374 gets merged.

Implements image gallery for products. Most logic is handled by `base_multi_image`.

@rafaelbn

### CODE REVIEW

**Comment by carlosdauden** [comment]
When I install the module product images are "losed"...
They should migrate to the new model?

**Comment by carlosdauden** [comment]
Functional test in local :+1:

File: product_multi_image/README.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+   :alt: License: AGPL-3
+
+===========================
+Multiple Images in Products
+===========================
+
+This module implements the possibility to have multiple images for a product,
+that it's to say, an image gallery.
+
+Usage
```
Comment (pedrobaeza) [inline product_multi_image/README.rst:20]: Add Installation section talking about the dependency to the base module and in which repo it is.

File: product_multi_image/hooks.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.addons.base_multi_image.hooks import post_init_hook_for_submodules
+
+
+def post_init_hook(cr, registry):
+    post_init_hook_for_submodules(cr, registry, "product.template", "image")
```
Comment (pedrobaeza) [inline product_multi_image/hooks.py]: Shouldn't this be `product.product` instead?

**Comment by pedrobaeza** [comment]
For the icon, maybe you can add a symbol/object to represent the product for distinguishing from the base module.

**Comment by andhit-r** [comment]
@Yajo i could not force rebuild runbot to test

**Comment by yajo** [comment]
All of the above fixed and/or documented. But this will :x: until https://github.com/OCA/server-tools/issues/385 is merged. It's a little patch, go there before please :blush:

**Comment by yajo** [comment]
https://github.com/OCA/server-tools/pull/385 merged. Let's see if you lose no more images, @carlos-incaser. :wink:

**Comment by atchuthan** [comment]
:+1: code reviewed

**Comment by pedrobaeza** [comment]
:+1:

**Comment by lukebranch** [comment]
@Yajo ,

Thanks for your work with this module. I'd like to check what the OCA's policy is on including OCA modules as dependencies in paid themes.

The website_multi_image module seems like it would be a vital component of most themes currently available for Odoo. Many have their own versions of website_multi_image included within the theme itself, however I think @Yajo's module handles this far better, and I believe it will quickly become the de-facto standard (for 8.0/9.0/Master).

I'm not interested in starting another endless licensing debate, i'm just curious to know what the OCA's policy is given the situation i've just described so that I can plan ahead and decide whether to include the multi-image module as a direct dependency or if I should declare it as 'compatible with' Yajo's modules.

My apologies for diverting from the topic of this issue thread. If there is a more appropriate place to ask this question I will remove this comment and ask wherever you suggest.

**Comment by pedrobaeza** [comment]
@lukebranch, indeed this is not the right place for this discussion, but in sort, you can't. If you derive from an AGPL module, you have to be AGPL too.

**Comment by lukebranch** [comment]
@pedrobaeza ,

Thanks for the confirmation.

**Comment by yajo** [comment]
Hi @lukebranch, of course you can as long as your license is [compatible with AGPL](http://www.gnu.org/licenses/license-list.html#GPLCompatibleLicenses). And [you are allowed to ask for money for your themes](http://www.gnu.org/licenses/gpl-faq.html#DoesTheGPLAllowDownloadFee).

**Comment by lukebranch** [comment]
@Yajo ,

Thanks for the confirmation, I look forward to testing the module in production soon and a few themes i'm working on.

By the way, for website_multi_image i've just come across a new GPU-accelerated slider that looks quite promising and is much more actively maintained than OwlCarousel here:

https://github.com/nolimits4web/Swiper

I haven't looked into it in detail yet, however it looks like it could work quite well for this purpose.

**Pull request merged**